source/develop/release-management/features/storage/gluster-multiple-bricks-per-storage.md

```diff
@@ -0,0 +1,74 @@
+feature_modules: engine/vdsm
+feature_status: Planning
+---
+
+# Support for multiple GlusterFS bricks on a block device(s)
+
+# Summary
+
+This feature introduces ability to create multiple GlusterFS bricks on the same block device(s).
+
+In case of shared block device, engine should create a LVM volume group on top of it and use LVM logical volumes as a brick backend devices. That approach will allow management of the local storage alongside with support for thin provisioning for the GlusterFS and SSD caching.
+
+## Owner
+
+*   Name: [Denis Chaplygin](https://github.com/akashihi)
+*   Email: <dchaplyg@redhat.com>
+
+## Current status
+
+*   Target Release: 4.3
+*   Status: Planning
+*   Last updated: December 19, 2017
+
+# Detailed description
+
+Current oVirt's GlusterFS brick/volume management implementation implies that bricks should occupy the whole block device, thus limiting possible number of bricks to the number of physically available block devices. That approach could be quite limiting, especially in case of small deployments or hyperconverged environments, when you have just one or two relatively big drives for you VM payloads.
+
+With this change we will introduce LVM management support into oVirt, thus allowing users to manage their local storage using oVirt UI and use LVM logical volumes as a block devices for GlusterFS bricks. Due to LVM ability it will also bring LVM specific features, such as thin provisioning and SSD caching.
+
+# Implementation details
+
+## Engine
+
+* New LVM specific entities should be added to represent managed LVM data.
+* Database access layer needs to be extended to store per-host managed LVM data
+* Engine should support querying LVM status and configuration on hosts, manage physical volumes, volume groups and logical volumes. This can be implemented using engine's Ansible integration, to not to pollute vdsm with that code.
+* Engine should support LVM thin pools and allow creation of thin logical volumes. Thin pool implementation details should be hidden from the user.
+* Alongside with thin pools, engine must support SSD caching configuration. Implementation details should be hidden from the user.
+* Thin pool support adds a 'second' size to the logical volumes (and, therefore, to gluster bricks, gluster volumes and storage domains running on top of them) - with thin pool every logical volume have a 'reported' size and 'physical' size. First one, 'reported' is a size of a logical volume, as reported by df command or similar. 'Physical' size is a size of thn pool itself. Obviously, they could differ and actual usage of both filesystem and thin pool may differ, so both sizes must be reported by vdsm to the engine, stored at the database and presented to the user.
+* It would be nice if engine will actually know type of each block device (solid/spinning) and will only allow solid drives to be used as SSD cache. We need that information to be reported by VDSM and stored at the database.
```

Comment:
sold/spinning is very easy to query per device and store.

Reply:
IT is nice, that we already have support for that query